src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -113,9 +125,61 @@ case class Cluster(clusterName: ClusterName,
   def projectNameString: String = s"${googleProject.string}/${clusterName.string}"
 }
 
+object MachineConfig {
+  implicit val machineConfigSemigroup = new Semigroup[MachineConfig] {
+    def combine(defined: MachineConfig, default: MachineConfig): MachineConfig = {
+      val minimumDiskSize = 100
+      defined.numberOfWorkers match {
+        case None | Some(0) => MachineConfig(Some(0), defined.masterMachineType.orElse(default.masterMachineType),
+                                             applyMinimumDiskSize(defined.masterDiskSize, default.masterDiskSize, minimumDiskSize))
+        case Some(numWorkers) if numWorkers < 2 => throw TooFewWorkersRequestedException(numWorkers)
+        case numWorkers => MachineConfig(numWorkers, defined.masterMachineType.orElse(default.masterMachineType),
+                                applyMinimumDiskSize(defined.masterDiskSize, default.masterDiskSize, minimumDiskSize),
+                                defined.workerMachineType.orElse(default.workerMachineType),
+                                applyMinimumDiskSize(defined.workerDiskSize, default.workerDiskSize, minimumDiskSize),
+                                defined.numberOfWorkerLocalSSDs.orElse(default.numberOfWorkerLocalSSDs),
+                                defined.numberOfPreemptibleWorkers.orElse(default.numberOfPreemptibleWorkers))
+      }
+    }
+  }
+
+  def applyMinimumDiskSize(definedDiskSize: Option[Int], defaultDiskSize: Option[Int], minimumDiskSize: Int): Option[Int] = {
+    definedDiskSize match {
```

Comment:
This lets you pass a default that's below the minimum. Is that OK?

An alternative may be:
```
definedDiskSize.orElse(defaultDiskSize).map(s => math.max(minimumDiskSize, s))
```